internal/provider/kubernetes/controller.go

```diff
@@ -325,32 +418,35 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, _ reconcile.Reques
 			}
 		}
 
-		// process envoy gateway secret refs
-		r.processEnvoyProxySecretRef(ctx, gwcResource)
-		gc := status.SetGatewayClassAccepted(
-			managedGC.DeepCopy(),
-			true,
-			string(gwapiv1.GatewayClassReasonAccepted),
-			status.MsgValidGatewayClass)
-		r.resources.GatewayClassStatuses.Store(utils.NamespacedName(gc), &gc.Status)
-
 		if len(gwcResource.Gateways) == 0 {
 			r.log.Info("No gateways found for accepted gatewayClass")
 
 			// If needed, remove the finalizer from the accepted GatewayClass.
 			if err := r.removeFinalizer(ctx, managedGC); err != nil {
+				if isTransientError(err) {
+					return reconcile.Result{}, err
+				}
+
 				r.log.Error(err, fmt.Sprintf("failed to remove finalizer from gatewayClass %s",
 					managedGC.Name))
-				continue
 			}
 		} else {
 			// finalize the accepted GatewayClass.
 			if err := r.addFinalizer(ctx, managedGC); err != nil {
+				if isTransientError(err) {
+					return reconcile.Result{}, err
+				}
+
+				// Failed to add finalizer should not affect the xDS cache update.
+	            // Log the error and continue processing.
 				r.log.Error(err, fmt.Sprintf("failed adding finalizer to gatewayClass %s",
 					managedGC.Name))
-				continue
 			}
 		}
+
+		// Set HasErrors to true if there are any non-transient errors during processing of the Gateway and Envoy Gateway resources.
+		// This will be used to determine whether the xDS cache should be updated or not.
+		gwcResource.HasErrors = (len(errs) > 0)
```

Comment:
Should we surface the non-transient errors from provider layer to the GatewayClass status?